File: crypto/src/main/java/net/corda/v5/crypto/merkle/MerkleTreeHashDigest.java
package net.corda.v5.crypto.merkle;

import net.corda.v5.crypto.DigestAlgorithmName;
import org.jetbrains.annotations.NotNull;

/**
 * Different use cases require different {@link MerkleTree} calculations.
 * {@link MerkleTreeHashDigest}s let us specify such implementations.
 */
public interface MerkleTreeHashDigest {
    
    /**
     * Get the algorithm name
     *
     * @return Specifies the digest algorithm.
     */
    @NotNull
    DigestAlgorithmName getDigestAlgorithmName();
}

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigest.kt
package net.corda.v5.crypto.merkle

import net.corda.v5.crypto.DigestAlgorithmName

/**
 * Different use cases require different [MerkleTree] calculations.
 * [MerkleTreeHashDigest]s let us specify such implementations.
 *
 *  @property digestAlgorithmName Specifies the digest algorithm.
 */

interface MerkleTreeHashDigest {
    val digestAlgorithmName: DigestAlgorithmName
}
